jetstream/consumer.go

```diff
@@ -318,8 +318,11 @@ func deleteConsumer(ctx context.Context, js *jetStream, stream, consumer string)
 }
 
 func validateConsumerName(dur string) error {
-	if strings.Contains(dur, ".") {
-		return fmt.Errorf("%w: %q", ErrInvalidConsumerName, dur)
+	if dur == "" {
+		return fmt.Errorf("%w: '%s'", ErrInvalidConsumerName, "name is required")
+	}
+	if strings.ContainsAny(dur, ">*. /\\") {
```

Comment:
This now is aligned with CLI and .js clients, but I'm afraid implementing this change could break some users.
Any thoughts @ripienaar ?

Reply:
Yeah, perhaps the only one the server wouldnt catch really is the space right?

Reply:
actually no - space would break the protocol there so that wouldnt have worked, I think its fine but we do need to call it out carefully

Reply:
we definately need to add space and wildcards, but `/` and `\` are not that obvious IMO.

Reply:
`/` and `\` are already rejected by the server with `API error: code=400 err_code=10128 description=Stream name can not contain path separators` (same for consumer names), so the only change here would be client-side validation instead of server-side. But I don't think we absolutely need it since this at least does not break protocol when put in subject.

Reply:
qq, when creating a consumer do we check for whether trying to create an filter subject like `foo.>.` or `foo.>.*`?

Reply:
yes ending with `.` need to be client side since it becomes a malformed subject so the server cannot match the JS API, `foo.>.*` looks like the server does handle

Reply:
I added validating filter subject